Add Cygwin to the CI

ethomson opened the pull request.

fd00 commented:
This action may be easy to implement Github Actions
- https://github.com/cygwin/cygwin-install-action

ethomson commented:
Yeah I saw that (and used it as an inspiration for our setup script). Thanks!

I want to keep everything in a setup script (if possible) so that it's easy to bootstrap a build on a local dev box but if there's a compelling reason to use a setup action instead I'm not opposed.

ethomson committed:
ci: add temporary cygwin build

Add a temporary Cygwin build to the PR CI while we work on Cygwin
compatibility.

We will move this to the nightlies when it is green.

ethomson committed:
ci: Cygwin builds for x86 as well

ethomson committed:
update cygwin for windows-2022 runners
